analyzers/src/SonarAnalyzer.Common/Helpers/KnownMethods.cs

```diff
@@ -87,26 +87,17 @@ public static bool IsStaticObjectEquals(this IMethodSymbol methodSymbol)
         && methodSymbol.Parameters.Length == 0
         && methodSymbol.ReturnType.Is(KnownType.System_String);
 
-    public static bool IsIDisposableDispose(this IMethodSymbol methodSymbol)
-    {
-        const string explicitName = "System.IDisposable.Dispose";
-        return methodSymbol != null
-            && (methodSymbol.Name == nameof(IDisposable.Dispose) || methodSymbol.Name == explicitName)
-            && methodSymbol.Arity == 0
-            && methodSymbol.ReturnsVoid
-            && methodSymbol.Parameters.Length == 0;
-    }
-
-    public static bool IsIAsyncDisposableDisposeAsync(this IMethodSymbol methodSymbol)
-    {
-        const string explicitNameAsync = "System.IAsyncDisposable.DisposeAsync";
-        return methodSymbol != null
-            && (methodSymbol.Name == "DisposeAsync" || methodSymbol.Name == explicitNameAsync)
-            && methodSymbol.ReturnType.Is(KnownType.System_Threading_Tasks_ValueTask)
-            && methodSymbol.Parameters.Length == 0
-            && ContainingInterface(methodSymbol) is { } containingInterface
-            && containingInterface.Is(KnownType.System_IAsyncDisposable);
-    }
+    public static bool IsIDisposableDispose(this IMethodSymbol methodSymbol) =>
```

Comment:
Both methods were refactored to use pattern matching. Functional changes:
- check whether the method is `static`
- check for ref struct types for `Dispose` invocations